drivers/sensor/mcp9600/mcp9600.c

```diff
@@ -64,23 +66,40 @@ static int mcp9600_sample_fetch(const struct device *dev, enum sensor_channel ch
 	uint8_t buf[2];
 	int ret;
 
-	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_AMBIENT_TEMP) {
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_AMBIENT_TEMP &&
+		chan != SENSOR_CHAN_DIE_TEMP) {
```

Comment:
Please correct me if I'm wrong, but It feels like you hijacked the DIE_TEMP channel because you needed something that resembled a temperature, and AMBIENT_TEMP was already taken? The proper way to do it would be to create a dedicated, sensor-specific channel.

Reply:
Well, there are a couple of things one could say here 🙂 

1. One could argue that indeed the thermocouple hot-junction temperature should not have been called `AMBIENT_TEMP`, although I also understand why it could have been named like this.
2. The cold-junction temperature is most probably (although not very clearly described so in the datasheet) indeed the die temperature of the IC, so one could argue that `DIE_TEMP` is the correct one. However, in the datasheet it is indeed mentioned as cold-junction or ambient temperature.
3. We already have an in-tree example, where the `DIE_TEMP` has been used for exactly similar case. I'm referring to MAX31855, which (at least in this regard) has the same functionality as MCP9600. See [here](https://github.com/zephyrproject-rtos/zephyr/blob/731ba72423481fbc8e9f0b5bdf321423bad82bb3/drivers/sensor/max31855/max31855.c#L85).

We could indeed choose to make it properly, perhaps by defining two different channels, one for the hot-junction temperature and one for the cold-junction temperature? Notice also, that the MAX6675 driver will also be affected in that case, although there only the hot-junction temperature is present.